### PR TITLE
Creating / Editing / Showing statute paragraph amendments

### PR DESCRIPTION
**Pull request by CatoTH**
A very basic first implementation of the specification for the statute paragraphs:
- If statute paragraphs are defined: When creating a motion, a checkbox appears to set this motion as a statute paragraph amendment. If the checkbox is activated, a SELECT appears, allowing to select the paragraph. If one is selected, the title and text below are pre-filled.
- After creating the statute amendment, the amendment is shown in simple diff view.

Questions for potential enhancements would be:
- Do statute amendments have line numbering? Or does the statute in itself have line numbers, for that matter? (If so, it would probably have to be continuous along the different paragraph objects)
- What views does such an amendment have? Diff, obviously. But also the "changed" view?
- How to handle the workflows? I'd be glad if that could be implemented by someone else, as I'm not very familiar with them.

### CODE REVIEW

File: client/src/app/site/motions/components/motion-detail/motion-detail.component.html [hunk 1]
```diff
@@ -268,11 +268,29 @@ <h3 translate>Voting</h3>
             </button>
         </div>
 
+        <!-- Selecting statute paragraphs for amendment -->
+        <div class="statute-amendment-selector" *ngIf="editMotion && statuteParagraphs.length > 0">
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/motion-detail/motion-detail.component.html]: Looks out of place here and moves the text and reason around.
I would move it into the meta data block and perhaps just use a `mat-select` without a checkbox

**Comment by CatoTH** [comment]
> What is the use case if you de-select the "Statute amendment" checkbox from an existing statute amendment motion? It seems like it was converted into a normal motion.

That would be up to discussion, if changing the statute paragraph afterwards should be supported. It sounds reasonable to just not provide this functionality alltogether.

> Furthermore, using this feature seems to substitute the while motion text with the text defied in the Statute paragraphs. Is this intended?

As far as I know, that was in the original specification that I got for this feature. @emanuelschuetze can probably clarify.

**Comment by CatoTH** [comment]
As for the config variables:
- `motions_amendments_enabled`: Ok, is done.
- `motions_amendments_prefix`: Hm, not sure here. Do statute amendments have the same prefixes and numbering system as regular amendments?
- `motions_amendments_text_mode`: Not sure here either, especially as the values from that variables do not really match the logic of statute amendments, esp. "Edit the whole motion text" and "Paragraph-based, Diff-enabled". "Empty text field" might be possible to implement, but then again, what would be the point of selecting a statute paragraph in the first place? :)

**Comment by CatoTH** [comment]
Instead of `motions_amendments_enabled`, the new variable `motions_statutes_enabled` is now used. It's currently in motions -> general, and I had to rebuild my database (configstore) for it to work.

**Comment by emanuelschuetze** [comment]
@tsiegleauq Please review and merge!

**Pull request merged**